scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -1080,6 +1377,11 @@ void BPF_STRUCT_OPS(rusty_set_weight, struct task_struct *p, u32 weight)
 			   taskc->weight, weight);
 
 	taskc->weight = weight;
+
+	if (fifo_sched)
+		return;
+
+	clamp_task_vtime(p, taskc, 0);
```

Comment:
I wonder if there is a case that a task is set_weight()-ed after it is enqueued.
If there are such cases, don't we need to re-order the task in the DSQ?
Otherwise, we don't need to calculate the deadline here?

Reply:
Hmm, yes, I think you're right that we don't need to set the deadline here. A task should never have `ops.set_weight()` called while it's enqueued in a DSQ. The core scheduler will first issue a dequeue before calling `p->sched_class->set_weight()`. So I think we can just rely on the enqueue path to set the min vtime and deadline. Thanks for catching this!